Treat AsyncReplies as first class messages in the IPCTestingAPI

gavin-apple opened the pull request:
#### a38ab639b9f37940f21f2abbadb592ba0ceb2ca3
<pre>
Treat AsyncReplies as first class messages in the IPCTestingAPI
<a href="https://bugs.webkit.org/show_bug.cgi?id=298388">https://bugs.webkit.org/show_bug.cgi?id=298388</a>
<a href="https://rdar.apple.com/158580728">rdar://158580728</a>

Reviewed by Alex Christensen.

This change exposes dispatched(From|To), and other properties, for AsyncReply
messages to the IPCTestingAPI. Previously, our model associated AsyncReplies
with a single global AsyncReply interface and not the original interface they belong to.
As a result, properties like dispatched(From|To) &amp; arguments weren&apos;t being plumbed
through.

This change updates our model so that we now associate all AsyncReply messages
with their correct interface. Messages are now marked as being an async
reply using a new MessageDescriptor property (isAsyncReply) and Message class generation
is now performed such that the properties can be correctly accessed from
our testing interfaces.

* LayoutTests/ipc/dispatched-from-to-exposed-to-ipc-testing-api.html:
* LayoutTests/ipc/isasyncreply-exposed-to-ipc-testing-api-expected.txt: Added.
* LayoutTests/ipc/isasyncreply-exposed-to-ipc-testing-api.html: Added.
* LayoutTests/ipc/replyarguments-exposed-to-ipc-testing-api-expected.txt: Added.
* LayoutTests/ipc/replyarguments-exposed-to-ipc-testing-api.html: Added.
* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::processIncomingMessage):
(IPC::Connection::dispatchMessage):
* Source/WebKit/Platform/IPC/Connection.h:
(IPC::Connection::waitForAsyncReplyAndDispatchImmediately):
* Source/WebKit/Platform/IPC/Decoder.h:
(IPC::Decoder::isAsyncReplyMessage const):
* Source/WebKit/Scripts/webkit/messages.py:
(receiver_enumerator_order_key):
(MessageEnumerator.__str__):
(generate_message_handler.collect_message_statements):
(generate_message_names_header):
(generate_message_names_implementation):
* Source/WebKit/Scripts/webkit/model.py:
(Message.__init__):
(generate_global_model):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::jsValueForArguments):
(IPC::messageArgumentDescriptions):
* Source/WebKit/Scripts/webkit/tests/MessageNames.cpp:
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
(IPC::isAsyncReply):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithCVPixelBuffer_ReceiveCVPixelBufferReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::name):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::ReceiveCVPixelBufferReply):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithImageData_ReceiveImageDataReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::ReceiveImageDataReply::name):
(Messages::TestWithImageData::ReceiveImageDataReply::ReceiveImageDataReply):
(Messages::TestWithImageData::ReceiveImageDataReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
(WebKit::TestWithLegacyReceiver::didReceiveMessage):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithLegacyReceiver_CreatePluginReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithLegacyReceiver_RunJavaScriptAlertReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithLegacyReceiver_GetPluginsReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithLegacyReceiver_InterpretKeyEventReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::CreatePluginReply::name):
(Messages::TestWithLegacyReceiver::CreatePluginReply::CreatePluginReply):
(Messages::TestWithLegacyReceiver::CreatePluginReply::encode):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::name):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::RunJavaScriptAlertReply):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::encode):
(Messages::TestWithLegacyReceiver::GetPluginsReply::name):
(Messages::TestWithLegacyReceiver::GetPluginsReply::GetPluginsReply):
(Messages::TestWithLegacyReceiver::GetPluginsReply::encode):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::name):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::InterpretKeyEventReply):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithSemaphore_ReceiveSemaphoreReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::name):
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::ReceiveSemaphoreReply):
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithStream_SendStringAsyncReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithStream_CallWithIdentifierReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendStringAsyncReply::name):
(Messages::TestWithStream::SendStringAsyncReply::SendStringAsyncReply):
(Messages::TestWithStream::SendStringAsyncReply::encode):
(Messages::TestWithStream::CallWithIdentifierReply::name):
(Messages::TestWithStream::CallWithIdentifierReply::CallWithIdentifierReply):
(Messages::TestWithStream::CallWithIdentifierReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithSuperclass_TestAsyncMessageReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithSuperclass_TestAsyncMessageWithNoArgumentsReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithSuperclass_TestAsyncMessageWithMultipleArgumentsReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithSuperclass_TestAsyncMessageWithConnectionReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::TestAsyncMessageReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageReply::TestAsyncMessageReply):
(Messages::TestWithSuperclass::TestAsyncMessageReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::TestAsyncMessageWithNoArgumentsReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::TestAsyncMessageWithMultipleArgumentsReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::TestAsyncMessageWithConnectionReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithValidatorMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithValidator_MessageWithReplyReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithValidatorMessages.h:
(Messages::TestWithValidator::MessageWithReplyReply::name):
(Messages::TestWithValidator::MessageWithReplyReply::MessageWithReplyReply):
(Messages::TestWithValidator::MessageWithReplyReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
(WebKit::TestWithoutAttributes::didReceiveMessage):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutAttributes_CreatePluginReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutAttributes_RunJavaScriptAlertReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutAttributes_GetPluginsReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutAttributes_InterpretKeyEventReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::CreatePluginReply::name):
(Messages::TestWithoutAttributes::CreatePluginReply::CreatePluginReply):
(Messages::TestWithoutAttributes::CreatePluginReply::encode):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::name):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::RunJavaScriptAlertReply):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::encode):
(Messages::TestWithoutAttributes::GetPluginsReply::name):
(Messages::TestWithoutAttributes::GetPluginsReply::GetPluginsReply):
(Messages::TestWithoutAttributes::GetPluginsReply::encode):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::name):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::InterpretKeyEventReply):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutUsingIPCConnection_MessageWithoutArgumentAndEmptyReplyReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutUsingIPCConnection_MessageWithoutArgumentAndReplyWithArgumentReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutUsingIPCConnection_MessageWithArgumentAndEmptyReplyReply&gt;):
(IPC::jsValueForDecodedMessage&lt;MessageName::TestWithoutUsingIPCConnection_MessageWithArgumentAndReplyWithArgumentReply&gt;):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessages.h:
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::MessageWithoutArgumentAndEmptyReplyReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::MessageWithoutArgumentAndReplyWithArgumentReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::MessageWithArgumentAndEmptyReplyReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::MessageWithArgumentAndReplyWithArgumentReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::encode):
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::emptyMessageWithReply):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/Shared/IPCTester.messages.in:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPC::messages):
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::EncoderDecoderTest::name):
* Tools/TestWebKitAPI/Tests/IPC/ConnectionTests.cpp:
(TestWebKitAPI::MockTestMessageWithConnection::name):
* Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp:
(TestWebKitAPI::MockTestMessageWithSignal::name):
* Tools/TestWebKitAPI/Tests/IPC/IPCTestUtilities.h:
(TestWebKitAPI::copyViaEncoder):
(TestWebKitAPI::MockTestMessage1::name):
(TestWebKitAPI::MockTestMessageWithAsyncReply1::name):
(TestWebKitAPI::MockTestMessageWithAsyncReply1::asyncMessageReplyName):

Canonical link: <a href="https://commits.webkit.org/299743@main">https://commits.webkit.org/299743@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb1345d81298f7d3119ac03c41923e5500efde54

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119959 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39652 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30303 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126283 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72032 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3ace26b6-b850-4f30-8787-620851a38163) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121835 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40347 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48229 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91093 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60408 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/19eac766-4846-4dc8-97e6-09a337e1b9f4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122911 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32225 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107561 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71649 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8f12ccda-2764-40cf-990d-87ace9d856c1) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/119316 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31256 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25667 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69923 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101700 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25855 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129205 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46879 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35542 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99719 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47245 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103750 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99564 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25295 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45011 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23026 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43500 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46741 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52447 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46207 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49556 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47893 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Scripts/webkit/tests/MessageNames.h:
```diff
@@ -266,6 +257,7 @@ struct MessageDescription {
     bool messageAllowedWhenWaitingForUnboundedSyncReply : 1;
     ProcessName dispatchedFrom;
     ProcessName dispatchedTo;
+    bool isAsyncReply;
```
nit: this could be with the other bools and have : 1

gavin-apple committed:
Treat AsyncReplies as first class messages in the IPCTestingAPI
https://bugs.webkit.org/show_bug.cgi?id=298388
rdar://158580728

Reviewed by Alex Christensen.

This change exposes dispatched(From|To), and other properties, for AsyncReply
messages to the IPCTestingAPI. Previously, our model associated AsyncReplies
with a single global AsyncReply interface and not the original interface they belong to.
As a result, properties like dispatched(From|To) & arguments weren't being plumbed
through.

This change updates our model so that we now associate all AsyncReply messages
with their correct interface. Messages are now marked as being an async
reply using a new MessageDescriptor property (isAsyncReply) and Message class generation
is now performed such that the properties can be correctly accessed from
our testing interfaces.

* LayoutTests/ipc/dispatched-from-to-exposed-to-ipc-testing-api.html:
* LayoutTests/ipc/isasyncreply-exposed-to-ipc-testing-api-expected.txt: Added.
* LayoutTests/ipc/isasyncreply-exposed-to-ipc-testing-api.html: Added.
* LayoutTests/ipc/replyarguments-exposed-to-ipc-testing-api-expected.txt: Added.
* LayoutTests/ipc/replyarguments-exposed-to-ipc-testing-api.html: Added.
* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::processIncomingMessage):
(IPC::Connection::dispatchMessage):
* Source/WebKit/Platform/IPC/Connection.h:
(IPC::Connection::waitForAsyncReplyAndDispatchImmediately):
* Source/WebKit/Platform/IPC/Decoder.h:
(IPC::Decoder::isAsyncReplyMessage const):
* Source/WebKit/Scripts/webkit/messages.py:
(receiver_enumerator_order_key):
(MessageEnumerator.__str__):
(generate_message_handler.collect_message_statements):
(generate_message_names_header):
(generate_message_names_implementation):
* Source/WebKit/Scripts/webkit/model.py:
(Message.__init__):
(generate_global_model):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::jsValueForArguments):
(IPC::messageArgumentDescriptions):
* Source/WebKit/Scripts/webkit/tests/MessageNames.cpp:
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
(IPC::isAsyncReply):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithCVPixelBuffer_ReceiveCVPixelBufferReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessages.h:
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::name):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::ReceiveCVPixelBufferReply):
(Messages::TestWithCVPixelBuffer::ReceiveCVPixelBufferReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithImageData_ReceiveImageDataReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h:
(Messages::TestWithImageData::ReceiveImageDataReply::name):
(Messages::TestWithImageData::ReceiveImageDataReply::ReceiveImageDataReply):
(Messages::TestWithImageData::ReceiveImageDataReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
(WebKit::TestWithLegacyReceiver::didReceiveMessage):
(IPC::jsValueForDecodedMessage<MessageName::TestWithLegacyReceiver_CreatePluginReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithLegacyReceiver_RunJavaScriptAlertReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithLegacyReceiver_GetPluginsReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithLegacyReceiver_InterpretKeyEventReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessages.h:
(Messages::TestWithLegacyReceiver::CreatePluginReply::name):
(Messages::TestWithLegacyReceiver::CreatePluginReply::CreatePluginReply):
(Messages::TestWithLegacyReceiver::CreatePluginReply::encode):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::name):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::RunJavaScriptAlertReply):
(Messages::TestWithLegacyReceiver::RunJavaScriptAlertReply::encode):
(Messages::TestWithLegacyReceiver::GetPluginsReply::name):
(Messages::TestWithLegacyReceiver::GetPluginsReply::GetPluginsReply):
(Messages::TestWithLegacyReceiver::GetPluginsReply::encode):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::name):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::InterpretKeyEventReply):
(Messages::TestWithLegacyReceiver::InterpretKeyEventReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithSemaphore_ReceiveSemaphoreReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessages.h:
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::name):
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::ReceiveSemaphoreReply):
(Messages::TestWithSemaphore::ReceiveSemaphoreReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithStream_SendStringAsyncReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithStream_CallWithIdentifierReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessages.h:
(Messages::TestWithStream::SendStringAsyncReply::name):
(Messages::TestWithStream::SendStringAsyncReply::SendStringAsyncReply):
(Messages::TestWithStream::SendStringAsyncReply::encode):
(Messages::TestWithStream::CallWithIdentifierReply::name):
(Messages::TestWithStream::CallWithIdentifierReply::CallWithIdentifierReply):
(Messages::TestWithStream::CallWithIdentifierReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithSuperclass_TestAsyncMessageReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithSuperclass_TestAsyncMessageWithNoArgumentsReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithSuperclass_TestAsyncMessageWithMultipleArgumentsReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithSuperclass_TestAsyncMessageWithConnectionReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessages.h:
(Messages::TestWithSuperclass::TestAsyncMessageReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageReply::TestAsyncMessageReply):
(Messages::TestWithSuperclass::TestAsyncMessageReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::TestAsyncMessageWithNoArgumentsReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithNoArgumentsReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::TestAsyncMessageWithMultipleArgumentsReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithMultipleArgumentsReply::encode):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::name):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::TestAsyncMessageWithConnectionReply):
(Messages::TestWithSuperclass::TestAsyncMessageWithConnectionReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithValidatorMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithValidator_MessageWithReplyReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithValidatorMessages.h:
(Messages::TestWithValidator::MessageWithReplyReply::name):
(Messages::TestWithValidator::MessageWithReplyReply::MessageWithReplyReply):
(Messages::TestWithValidator::MessageWithReplyReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
(WebKit::TestWithoutAttributes::didReceiveMessage):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutAttributes_CreatePluginReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutAttributes_RunJavaScriptAlertReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutAttributes_GetPluginsReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutAttributes_InterpretKeyEventReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessages.h:
(Messages::TestWithoutAttributes::CreatePluginReply::name):
(Messages::TestWithoutAttributes::CreatePluginReply::CreatePluginReply):
(Messages::TestWithoutAttributes::CreatePluginReply::encode):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::name):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::RunJavaScriptAlertReply):
(Messages::TestWithoutAttributes::RunJavaScriptAlertReply::encode):
(Messages::TestWithoutAttributes::GetPluginsReply::name):
(Messages::TestWithoutAttributes::GetPluginsReply::GetPluginsReply):
(Messages::TestWithoutAttributes::GetPluginsReply::encode):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::name):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::InterpretKeyEventReply):
(Messages::TestWithoutAttributes::InterpretKeyEventReply::encode):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessageReceiver.cpp:
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutUsingIPCConnection_MessageWithoutArgumentAndEmptyReplyReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutUsingIPCConnection_MessageWithoutArgumentAndReplyWithArgumentReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutUsingIPCConnection_MessageWithArgumentAndEmptyReplyReply>):
(IPC::jsValueForDecodedMessage<MessageName::TestWithoutUsingIPCConnection_MessageWithArgumentAndReplyWithArgumentReply>):
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnectionMessages.h:
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::MessageWithoutArgumentAndEmptyReplyReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndEmptyReplyReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::MessageWithoutArgumentAndReplyWithArgumentReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithoutArgumentAndReplyWithArgumentReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::MessageWithArgumentAndEmptyReplyReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndEmptyReplyReply::encode):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::name):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::MessageWithArgumentAndReplyWithArgumentReply):
(Messages::TestWithoutUsingIPCConnection::MessageWithArgumentAndReplyWithArgumentReply::encode):
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::emptyMessageWithReply):
* Source/WebKit/Shared/IPCTester.h:
* Source/WebKit/Shared/IPCTester.messages.in:
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPC::messages):
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::EncoderDecoderTest::name):
* Tools/TestWebKitAPI/Tests/IPC/ConnectionTests.cpp:
(TestWebKitAPI::MockTestMessageWithConnection::name):
* Tools/TestWebKitAPI/Tests/IPC/EventTests.cpp:
(TestWebKitAPI::MockTestMessageWithSignal::name):
* Tools/TestWebKitAPI/Tests/IPC/IPCTestUtilities.h:
(TestWebKitAPI::copyViaEncoder):
(TestWebKitAPI::MockTestMessage1::name):
(TestWebKitAPI::MockTestMessageWithAsyncReply1::name):
(TestWebKitAPI::MockTestMessageWithAsyncReply1::asyncMessageReplyName):

Canonical link: https://commits.webkit.org/299743@main
